apps/cowswap-frontend/src/legacy/components/ErrorBoundary/index.tsx

```diff
@@ -1,2 +1,152 @@
+    position: relative;
+  }
+`
+const FooterWrapper = styled(HeaderWrapper)`
+  z-index: 1;
+  flex-grow: 1;
+  width: 100%;
+  position: relative;
+  top: auto;
+`
+
+async function updateServiceWorker(): Promise<ServiceWorkerRegistration> {
+  const ready = await navigator.serviceWorker.ready
+  // the return type of update is incorrectly typed as Promise<void>. See
+  // https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/update
+  return ready.update() as unknown as Promise<ServiceWorkerRegistration>
+}
+
+export default class ErrorBoundary extends React.Component<PropsWithChildren, ErrorBoundaryState> {
+  constructor(props: PropsWithChildren) {
+    super(props)
+    this.state = { error: null }
+  }
+
+  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
+    updateServiceWorker()
+      .then(async (registration) => {
+        // We want to refresh only if we detect a new service worker is waiting to be activated.
+        // See details about it: https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle
+        if (registration?.waiting) {
+          await registration.unregister()
+
+          // Makes Workbox call skipWaiting(). For more info on skipWaiting see: https://developer.chrome.com/docs/workbox/handling-service-worker-updates/
+          registration.waiting.postMessage({ type: 'SKIP_WAITING' })
+
+          // Once the service worker is unregistered, we can reload the page to let
+          // the browser download a fresh copy of our app (invalidating the cache)
+          window.location.reload()
+        }
+      })
+      .catch((error) => {
+        console.error('Failed to update service worker', error)
+      })
+    return { error }
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    sendError(error, errorInfo)
+  }
+
+  render() {
+    return (
+      <Sentry.ErrorBoundary
+        showDialog={false}
+        fallback={({ error: sentryError }) => {
+          document.body.classList.remove('noScroll')
+          const { error: localError } = this.state
+          const error = localError || sentryError
+
+          const isChunkLoadError =
+            error?.name === 'ChunkLoadError' || error?.message.includes('Failed to fetch dynamically imported module')
+
+          return (
+            <AppWrapper>
+              <HeaderWrapper>
+                <HeaderRow marginRight="0">
+                  <a href={Routes.HOME}>
+                    <UniIcon>
+                      <LogoImage />
+                    </UniIcon>
+                  </a>
+                </HeaderRow>
+              </HeaderWrapper>
+              <Wrapper>{isChunkLoadError ? <ChunkLoadError /> : <ErrorWithStackTrace error={error} />}</Wrapper>
+              <FooterWrapper>
+                <Footer />
+              </FooterWrapper>
+            </AppWrapper>
+          )
+        }}
+      >
+        {this.props.children}
+      </Sentry.ErrorBoundary>
```

Comment:
This is the main change.
Wrapped the whole component response in Sentry.ErrorBoundary and put the fallback inside of it